### PR TITLE
fix(scrollable-block): ensure border-styling is applied when appropriate to do so

### PR DESCRIPTION
**Pull request by mihai-albu-sage**
### Proposed behaviour

Border radius should be 0px on all corners when the last menu item does not exceed the bottom of the scrollable-block or the submenu itself. 0px, 0px, 0px, 8px should only be applied when the last menu item exceeds the bottom of the submenu block. 

Screenshot 2024-04-08 082650
Screenshot 2024-04-08 082719

### Current behaviour

border-radius styling was being set incorrectly

previous behaviour Screenshot 2024-04-08 083135

### Checklist

- [X] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [X] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Go to the [Scrollable submenu with parent](https://carbon.sage.com/?path=/docs/menu--scrollable-submenu-with-parent)
- Open the Submenu (Menu Item Three)
- In the Search component type ap and you see it returns two results. apple and grapefruit
- grapefruit has incorrect styling as you’ll be able to see the bottom left border has a 8px radius. It should be 0px in this case.
- Only if the scrollbar is present should the last menu item in the scrollable block have an 8px bottom left border radius applied.

### CODE REVIEW

**Review by DipperTheDan — COMMENTED** [review]
Good job on this. Not an easy component to work on. Only issue I spotted was this. If you add another `Menu Item` after the scrollable block, it has this styling which is incorrect. Message me if you want to see the code I used for this. 

Screenshot 2024-04-11 at 11 29 13

**Comment by mihai-albu-sage** [comment]
Since the recent updates to the submenu component, a more efficient solution is to implement the fix for this issue directly into the submenu component.

There was only one case which the submenu component was not taking into consideration which is if the last segment is a scrollable block and in this case to check if the last menu item in this block is visible or not and take that into consideration when applying the radius changes.

Scrollable block with parent - on last position



Scrollable block with parent - not on last position



This does not affect the Menu Test which still render correctly:

File: src/components/menu/__internal__/submenu/submenu.component.tsx [hunk 1]
```diff
@@ -200,18 +210,36 @@ const Submenu = React.forwardRef<
...
+            lastMenuItemInSegmentBlock.getBoundingClientRect().bottom <
+            lastSegmentBlock.getBoundingClientRect().bottom;
+        }
+
         // Check if the last item in the segment block is the same as the last MenuItem in the submenu
         const menuItemsMatch =
           !!lastMenuItemInSegmentBlock &&
           lastMenuItemInSegmentBlock === lastMenuItem;
 
+        // Applies the focus radius to the last segment block
```
Comment (DipperTheDan) [inline src/components/menu/__internal__/submenu/submenu.component.tsx:227]: **suggestion(non-blocking):** This comment needs changing as the `applyFocusRadius` is being used in the `StyledBox` of the `StyledScrollableBlock`. I'd say it should be changed to reflect that.

File: src/components/menu/menu.pw.tsx [hunk 2]
```diff
@@ -2393,6 +2394,47 @@ test.describe(
...
+      await subMenu.hover();
+      const searchInput = searchDefaultInput(page);
+      await searchInput.fill("app");
+      const scrollableBlock = scrollBlock(page);
+      await expect(scrollableBlock).toHaveCSS(
+        "border-radius",
+        "0px 0px 0px 8px"
+      );
+      const scrollableItem = scrollBlock(page).locator("a").last();
+      await expect(scrollableItem).toHaveCSS("border-radius", "0px");
```
Comment (DipperTheDan) [inline src/components/menu/menu.pw.tsx:2413]: **nitpick:** minor formatting stuff

```suggestion
      const scrollableBlock = scrollBlock(page);
      
      await expect(scrollableBlock).toHaveCSS(
        "border-radius",
        "0px 0px 0px 8px"
      );
      
      const scrollableItem = scrollBlock(page).locator("a").last();
      
      await expect(scrollableItem).toHaveCSS("border-radius", "0px");
```

File: src/components/menu/menu.pw.tsx [hunk 3]
```diff
@@ -2393,6 +2394,47 @@ test.describe(
...
+      await expect(scrollableBlock).toHaveCSS(
+        "border-radius",
+        "0px 0px 0px 8px"
+      );
+      const scrollableItem = scrollBlock(page).locator("a").last();
+      await expect(scrollableItem).toHaveCSS(
+        "border-radius",
+        "0px 0px 0px 8px"
+      );
+    });
```
Comment (DipperTheDan) [inline src/components/menu/menu.pw.tsx:2436]: **nitpick:** minor formatting stuff
```suggestion
      const scrollableBlock = scrollBlock(page);
      
      await expect(scrollableBlock).toHaveCSS(
        "border-radius",
        "0px 0px 0px 8px"
      );
      
      const scrollableItem = scrollBlock(page).locator("a").last();
      
      await expect(scrollableItem).toHaveCSS(
        "border-radius",
        "0px 0px 0px 8px"
      );
    });
```

Comment (mihai-albu-sage) [reply src/components/menu/menu.pw.tsx]: done.

File: src/components/menu/menu.pw.tsx (hunk 2 above)
Comment (mihai-albu-sage) [reply src/components/menu/menu.pw.tsx]: done.

File: src/components/menu/__internal__/submenu/submenu.component.tsx (hunk 1 above)
Comment (mihai-albu-sage) [reply src/components/menu/__internal__/submenu/submenu.component.tsx]: done.

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Pull request merged**